### PR TITLE
add unique_consecutive_op

### PR DESCRIPTION
**Pull request by firestonelib**
### PR types

New features
### PR changes

OPs
### Describe

add unique_consecutive op for support to eliminate all but the first element from every consecutive group of equivalent elements.

46e734bf4a7149be5a298eff1f8f230a

### CODE REVIEW

File: python/paddle/tensor/__init__.py [hunk 1]
```diff
@@ -88,6 +88,7 @@
 from .manipulation import stack  # noqa: F401
 from .manipulation import strided_slice  # noqa: F401
 from .manipulation import unique  # noqa: F401
+from .manipulation import unique_consecutive
```
Comment (TCChenlong) [inline python/paddle/tensor/__init__.py]: 需要按格式，加上 “# noqa: F401”

File: python/paddle/tensor/__init__.py [hunk 2]
```diff
@@ -331,6 +332,7 @@
            'strided_slice',
            'transpose',
            'unique',
+           'unique_consecutive'
```
Comment (TCChenlong) [inline python/paddle/tensor/__init__.py]: 少了 `,`

File: python/paddle/tensor/manipulation.py [hunk 3]
```diff
@@ -710,6 +710,106 @@ def squeeze_(x, axis=None, name=None):
...
+                       return_counts=False,
+                       axis=None,
+                       dtype="int64",
+                       name=None):
+    r"""
+    Eliminates all but the first element from every consecutive group of equivalent elements.
+    .. note:: This function is different from :func:`paddle.unique` in the sense that this function
+        only eliminates consecutive duplicate values. This semantics is similar to `std::unique`
+        in C++.
+    Args:
```
Comment (TCChenlong) [inline python/paddle/tensor/manipulation.py:732]: Args 前需要加空行，Returns:、Example: 同理

File: python/paddle/tensor/manipulation.py [hunk 4]
```diff
@@ -710,6 +710,106 @@ def squeeze_(x, axis=None, name=None):
...
 
+def unique_consecutive(x,
+                       return_inverse=False,
+                       return_counts=False,
+                       axis=None,
+                       dtype="int64",
+                       name=None):
+    r"""
+    Eliminates all but the first element from every consecutive group of equivalent elements.
+    .. note:: This function is different from :func:`paddle.unique` in the sense that this function
```
Comment (TCChenlong) [inline python/paddle/tensor/manipulation.py:729]: 这里也需要加一行空行

File: python/paddle/tensor/manipulation.py [hunk 5]
```diff
@@ -710,6 +710,106 @@ def squeeze_(x, axis=None, name=None):
...
+        only eliminates consecutive duplicate values. This semantics is similar to `std::unique`
+        in C++.
+    Args:
+        x(Tensor): the input tensor, it's data type should be float32, float64, int32, int64.
+        return_inverse(bool, optional): If True, also return the indices for where elements in
```
Comment (TCChenlong) [inline python/paddle/tensor/manipulation.py:734]: Else XXX。Default is False。

File: python/paddle/tensor/manipulation.py [hunk 6]
```diff
@@ -710,6 +710,106 @@ def squeeze_(x, axis=None, name=None):
...
+            the original input ended up in the returned unique consecutive tensor.
+        return_counts(bool, optional): If True, also return the counts for each unique consecutive element.
```
Comment (TCChenlong) [inline python/paddle/tensor/manipulation.py:736]: Else XXX。Default is False。

**Review by TCChenlong — COMMENTED** [review]
需要在 https://github.com/PaddlePaddle/docs 补充中文文档

File: python/paddle/tensor/manipulation.py (hunk 4 above)
Comment (firestonelib) [reply python/paddle/tensor/manipulation.py:729]: DONE

File: python/paddle/tensor/manipulation.py (hunk 3 above)
Comment (firestonelib) [reply python/paddle/tensor/manipulation.py:732]: DONE

File: python/paddle/tensor/manipulation.py (hunk 5 above)
Comment (firestonelib) [reply python/paddle/tensor/manipulation.py:734]: DONE

File: python/paddle/tensor/manipulation.py (hunk 6 above)
Comment (firestonelib) [reply python/paddle/tensor/manipulation.py:736]: DONE

File: python/paddle/tensor/__init__.py (hunk 1 above)
Comment (firestonelib) [reply python/paddle/tensor/__init__.py]: DONE

File: python/paddle/tensor/__init__.py (hunk 2 above)
Comment (firestonelib) [reply python/paddle/tensor/__init__.py]: DONE

**Review by TCChenlong — DISMISSED** [review]

**Review by jzhang533 — DISMISSED** [review]
lgtm

**Review by TCChenlong — APPROVED** [review]

**Review by jzhang533 — APPROVED** [review]
lgtm

File: python/paddle/tensor/manipulation.py [hunk 7]
```diff
@@ -717,6 +717,112 @@ def squeeze_(x, axis=None, name=None):
...
+        tuple: (out, inverse, counts). `out` is the unique consecutive tensor for `x`. `inverse` is provided only if `return_inverse` is True. `counts` is provided only if `return_counts` is True.
+
+    Example:
+        .. code-block:: python
+
+            import paddle 
+
+            x = paddle.to_tensor([1, 1, 2, 2, 3, 1, 1, 2])
+            output = paddle.unique_consecutive(x) # 
+            np_output = output.numpy() # [1 2 3 1 2]
```
Comment (XiaoguangHu01) [inline python/paddle/tensor/manipulation.py:755]: 这里不用返回numpy，参考paddle.add，用下面这种写法就行：
z = paddle.add(x, y)
print(z)  # [3., 8., 6. ]

**Review by XiaoguangHu01 — APPROVED** [review]
LG API

File: paddle/fluid/operators/unique_consecutive_op.cc [hunk 8]
```diff
@@ -0,0 +1,142 @@
+/* Copyright (c) 2019 PaddlePaddle Authors. All Rights Reserved.
```
Comment (zhiqiu) [inline paddle/fluid/operators/unique_consecutive_op.cc:1]: 2019->2021

**Review by zhiqiu — APPROVED** [review]
lgtm for op_function_generator

**Review by fuyinno4 — APPROVED** [review]

**Review by raindrops2sea — APPROVED** [review]
LGTM

**Pull request merged**